test/functional/api/command_spec.lua

```diff
@@ -65,26 +65,19 @@ describe('nvim_get_commands', function()
     local cmd3 = { addr=NIL,         bang=false, bar=true,  complete=NIL,      complete_arg=NIL,         count=NIL,  definition='call \128\253R3_ohyeah()',      name='Cmd3',    nargs='0', range=NIL,  register=false, keepscript=false, script_id=3, }
     local cmd4 = { addr=NIL,         bang=false, bar=false, complete=NIL,      complete_arg=NIL,         count=NIL,  definition='call \128\253R4_just_great()',  name='Cmd4',    nargs='0', range=NIL,  register=true,  keepscript=false, script_id=4, }
     source([[
-      let s:foo = 1
       command -complete=custom,ListUsers -nargs=+ Finger !finger <args>
     ]])
     eq({Finger=cmd1}, meths.get_commands({builtin=false}))
     command('command -nargs=1 -complete=dir -addr=arguments -count=10 TestCmd pwd <args>')
     eq({Finger=cmd1, TestCmd=cmd0}, meths.get_commands({builtin=false}))
 
     source([[
-      function! s:foo() abort
-      endfunction
       command -bang -nargs=* Cmd2 call <SID>foo(<q-args>)
     ]])
     source([[
-      function! s:ohyeah() abort
-      endfunction
       command -bar -nargs=0 Cmd3 call <SID>ohyeah()
     ]])
     source([[
-      function! s:just_great() abort
-      endfunction
```

Comment:
These changes still fail on latest master, so https://github.com/neovim/neovim/pull/28486 and related improvements by @zeertzjq don't address the goal of this PR.

Reply:
This PR's pretty stale so most of it'll need to be redone; I do plan on getting back to it at _some_ point though. (No ETA; somebody else can take over if they'd like in the meantime :grin:)

That said, I'd maybe do some things differently if I were to revive it. E.g: the `SID_TRACE_*` stuff from this PR looks dumb (pretty sure it has edge cases too :innocent:).